src/oc/cygwinprt.cpp

```diff
@@ -6,9 +6,9 @@
 #if !defined(__MINGW32__)
 #include "system.cpp"
 #define my_off64_t loff_t
-#else
+#else  // __MINGW32__
 #define my_off64_t off64_t
-#endif
+#endif  // __MINGW32__
 
 #include "mswinprt.cpp"
```

Comment:
Just a note on including cpp into cpp, this reminds me of autootools :) 
`cygwinprt.cpp` looks very slim now, maybe we can refactor it out.

Reply:
I agree. Given the labyrinthine nature of the ```#if``` for ```CYGWIN```, ```MINGW```, and ```__MINGW32```, the fact that one of the first things in mswinprt.cpp is
```
#if defined(MINGW)
#define CYGWIN
#endif
```
and the likelihood that we can abandon ```CYGWIN``` altogether (and so remove the private copy of ```system.cpp``` as well) it seems reasonable to reserve all this to a separate pull request.